src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Patterns.cs

```diff
@@ -48,12 +48,16 @@ BoundExpression TranslatePattern(BoundExpression input, BoundPattern pattern)
                         input = _factory.Local(temp);
                         for (int i = 0; i < pat.Subpatterns.Length; i++)
                         {
-                            var subProperty = pat.Subpatterns[i].Property;
+                            var subProperty = (pat.Subpatterns[i].Member as BoundPropertyPatternMember)?.MemberSymbol;
```

Comment:
There is an open LDM issue for how this lookup should work. When we have more guidance we can complete all the specified scenarios.